src/main/java/core/basesyntax/ShopServiceImpl.java

```diff
@@ -0,0 +1,27 @@
+package core.basesyntax;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ShopServiceImpl implements ShopService {
+    private final OperationStrategy operationStrategy;
+    private final Map<String, Integer> storage = new HashMap<>();
```

Comment:
According to the encapsulation principle, the storage map should be private and not be exposed directly. Consider providing methods to interact with it rather than making it accessible.